### DISCUSSION TITLE
Async validations in zod

### DISCUSSION THREAD

**Discussion by danielkellyio** [formkit/formkit#1386] [Ideas]
I'd like to do async validations in the zod plugin with the .refine method something like this:
```
z
    .string()
    .email()
    .refine(
      async (email) => {
        const res = await $fetch("/api/user-already-exists", {
          method: "GET",
          query: { email },
        });
        return !res.exists;
      },
      {
        message: "Email already exists",
      }
    ),
```

Currently this is impossible since the `createZodPlugin` function uses `safeParse` instead of `safeParseAsync`. 

Are there any downsides to changing this?

**Comment by justin-schroeder** [comment]
I dont think there is any reason why we cant use `safeParseAsync`. @andrew-boyd?
